### PR TITLE
Clarify the FOR loop documentation

### LINKED ISSUE #1003: Documentation of `FOR` is insufficient
The documentation of `FOR` isn't explicit about the meaning of its _stop_ argument. The documentation currently says that the value "increments from _start_ to _stop_", but in actuality `FOR` breaks right _before_ the value would reach _stop_. This should be documented explicitly — right now the documentation improperly assumes familiarity with Python, as this is only made clear by comparing the semantics with Python's `range` generator.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1003

### CODE REVIEW

**Comment by Rangi42** [comment]
@aaaaaa123456789 Does this clear up the `FOR` loop meaning?

File: man/rgbasm.5 [hunk 1]
```diff
@@ -1804,11 +1804,20 @@ The
...
-.Ar stop .
+.Ar stop ,
+i.e. it covers the half-open range from
+.Ar start
+(inclusive) to
+.Ar stop
+(exclusive).
+Any changes to the value within the
+.Ic FOR
+loop will not affect its subsequent updated value.
```
Comment (ISSOtm) [inline man/rgbasm.5:1815]:
```suggestion
The variable
.Ar V
will be made equal this value, as if by assignment, at the beginning of each new loop; any changes made to it within the loop's body will be overwritten.
Thus,
.Ar V
needs not exist before any iteration, but it must be a variable
.Pq Sx Variables
if it does.
```

**Review by ISSOtm — CHANGES_REQUESTED** [review]
This does not cover all edge cases, but I really couldn't find a simple model to explain those, so it'll have to do. (And I tried.)

File: man/rgbasm.5 [hunk 2]
```diff
@@ -1804,11 +1804,28 @@ The
...
+.Ar V
+will be assigned this value at the beginning of each new iteration; any changes made to it within the
+.Ic FOR
+loop's body will be overwritten.
+So if the symbol
+.Ar V
+is already defined before the
+.Ic FOR
+loop begins, it must be a variable
+.Pq Sx Variables .
```
Comment (ISSOtm) [inline man/rgbasm.5:1823]:
```suggestion
So the symbol
.Ar V
need not be already defined before any iterations of the
.Ic FOR
loop, but it must be a variable
.Pq Sx Variables
if so.
```
This clarifies that the symbol is defined at the beginning of *every* iteration, and that it must either not exist or be a variable as well. (Might be worth pointing out that the symbol stays defined afterwards, though the example can suffice for that.)

**Review by ISSOtm — CHANGES_REQUESTED** [review]

**Comment by ISSOtm** [comment]
This also does not fully close #1003 due to the aforementioned range edge cases with ridiculously large values. I think the issue should be kept open (and maybe the OP edited) to keep track of that.

File: man/rgbasm.5 (hunk 2 above)
Comment (Rangi42) [reply man/rgbasm.5:1823]: Okay.

**Comment by Rangi42** [comment]
> This also does not fully close #1003 due to the aforementioned range edge cases with ridiculously large values. I think the issue should be kept open (and maybe the OP edited) to keep track of that.

Let's make a separate issue for that; it sounds like there are more edge cases that the current issue doesn't cite. #1003

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**